src/pcre2_compile.c

```diff
@@ -1900,17 +1903,20 @@ else
 
       /* As we know we are at a digit, the only possible error from
       read_number() is a number that is too large to be a group number. In this
-      case we fall through handle this as not a group reference. If we have
-      read a small enough number, check for a back reference.
+      case we treat the group number as too-large (since it may be larger than
+      INT_MAX we cannot return it for the caller to check).
 
       \1 to \9 are always back references. \8x and \9x are too; \1x to \7x
       are octal escapes if there are not that many previous captures. */
 
-      if (read_number(&ptr, ptrend, -1, INT_MAX/10 - 1, 0, &s, errorcodeptr) &&
-          (s < 10 || oldptr[-1] >= CHAR_8 || s <= (int)cb->bracount))
+      if (!read_number(&ptr, ptrend, -1, MAX_GROUP_NUMBER, 0, &s, errorcodeptr))
+        {
+        s = INT_MAX;
+        }
+      if (s < 10 || oldptr[-1] >= CHAR_8 || s <= (int)cb->bracount)
```

Comment:
Values with `s < 10` and `s <= bracount` are unaffected. This fix only affects the escapes `\8xxxx` and `\9xxxx` which must be interpreted as backreferences, regardless of whether read_number succeeds or fails.